### DISCUSSION TITLE
Formula 1 Power BI Dashboard

### DISCUSSION THREAD

**Discussion by AntonioSchez32** [f1db/f1db#21] [Q&A] [answered]
Hi,

I'm planning on using different databases for a Power BI dashboard. I have looked Ergast DB, Fast F1 (which uses F1 live timing along with Ergast) and this one.

But I have seen that IDs for drivers, constructors, et cetera are different, but I think it's possible to merge data using other fields. Although I have to look into that.

Do you think it is possible using both databases since Ergast DB and Fast F1 have lap times and other info and Only F1 DB complement this info with the fastest laps, more detailed qualifying info, engine manufacturers, tyre manufacturers, etc.

Also, I'm worried about maintenance, so I would like to know if you are planning on updating this like you are doing at the moment. 

Thank you for your time.

**Comment by marceloverdijk** [comment]
Hi Antonio, 

Thanks for reaching out!
Yes I will be updating Only F1 DB after every race, since I'm doing for the past 2 seasons.

Adding lap times to Only F1 DB would be a great feature; I'm not aware of Fast F1 but will have a look at it over the holiday season.

What other data are you missing in Only F1 DB that Ergast DB and Fast F1 do have?

Regarding the IDs probably best to make some kind of mapping. It's very unlikely these would change anyway. So it would only mean adding a new mapping on your side if there is a new Driver, Constructor etc added to the database.

Does this help?

**Reply by marceloverdijk** [reply]
Btw, I just noticed the [Ergast API](https://ergast.com/mrd/) is deprecated and will be updated until the end of the 2024 season and then be shutdown at the end of the year.

For what it's worth, my intention with Only F1 was always to provide a API (REST + GraphQL) but I have no release date yet.

**Reply by AntonioSchez32** [reply]
Here it is the entity relationship diagram that is provided on the website. I'm going to figure out the best way to merge both datasets if possible.

![Ergast_db](https://github.com/onlyf1com/onlyf1-db/assets/72667525/8a78fae4-6edf-4cbc-bcb5-94e77eacf882)

Ergast DB is being deprecated at the end of 2024. But I think Fast F1 API people are taking over the project since they use it for their own API. There is some interesting information here on Fast F1 (https://docs.fastf1.dev/core.html) such as telemetry lap (tyres, tyre age, etc.) since 2018.

I think Only F1 DB is more complete (Fastest laps, Qualifying results, Starting Grid, Sprint Shootouts). But Ergast DB and Fast F1 API have Lap times since 1996 for the race. So it's worth to give it a go.

This is for my final degree project. I want to import CSV files along with Ergast DB CSV files and some Fast F1 telemetry into Power BI. Do some transformations, maybe add some details such as flags or photos for drivers and teams. Then, debug the model and add some measures. And, hopefully, in the end make some interesting graphics.

**Comment by AntonioSchez32** [comment, marked as answer]
> Btw, I just noticed the [Ergast API](https://ergast.com/mrd/) is deprecated and will be updated until the end of the 2024 season and then be shutdown at the end of the year.
> 
> For what it's worth, my intention with Only F1 was always to provide a API (REST + GraphQL) but I have no release date yet.

Wow, I have just downloaded the project and build it with Gradle and it's even better than I expected. Much more detailed data and statistics, I don't even have to calculate podiums and that kind of things with DAX. I still have to make some DAX for Head to head against teammate and that kind of stuff, but it will save me some headaches.

I still have to figure out how to include race lap times since 1996 to 2018, and telemetry lap times since 2018 to 2023, but it's just a tiny part compared to that. I'll keep some updates.
